### PR TITLE
chore: throw if more than one network config is passed

### PR DESCRIPTION
**Pull request by weboko**
## Problem

Users can provide more than one network config.

## Solution

Throw

## Notes

- Related to https://github.com/waku-org/js-waku/issues/2027

### CODE REVIEW

File: packages/sdk/src/utils/libp2p.ts [hunk 1]
```diff
@@ -128,6 +117,37 @@ export async function createLibp2pAndUpdateOptions(
...
+  if (options.contentTopics) {
+    options.shardInfo = { contentTopics: options.contentTopics };
+  }
+
+  const shardInfo = options.shardInfo
+    ? ensureShardingConfigured(options.shardInfo)
+    : undefined;
+
+  options.pubsubTopics = shardInfo?.pubsubTopics ??
+    options.pubsubTopics ?? [DefaultPubsubTopic];
```
[the remark this answers is not in the document]
Comment (weboko) [reply packages/sdk/src/utils/libp2p.ts:146]: I think usage of `??` is the most convenient at this point as each of operators would require a condition

File: packages/sdk/src/utils/libp2p.ts [hunk 2]
```diff
@@ -92,24 +92,13 @@ export async function defaultLibp2p(
...
-  options.pubsubTopics = shardInfo?.pubsubTopics ??
-    options.pubsubTopics ?? [DefaultPubsubTopic];
+  const shardInfo = configureNetworkOptions(options);
 
   const libp2pOptions = options?.libp2p ?? {};
   const peerDiscovery = libp2pOptions.peerDiscovery ?? [];
 
   if (options?.defaultBootstrap) {
-    peerDiscovery.push(...defaultPeerDiscoveries(options.pubsubTopics));
+    peerDiscovery.push(...defaultPeerDiscoveries(options.pubsubTopics!));
```
[the remark this answers is not in the document]
Comment (weboko) [reply packages/sdk/src/utils/libp2p.ts:101]: It is due to update happening inside of a function so TS don't pick it up and complains. 
To address this I think we should fully rework the approach to options and my proposal is here with draft PR https://github.com/waku-org/js-waku/issues/2073

File: packages/tests/tests/filter/utils.ts [hunk 3]
```diff
@@ -92,11 +92,15 @@ export async function runMultipleNodes(
...
     libp2p: {
       addresses: { listen: ["/ip4/0.0.0.0/tcp/0/ws"] }
-    },
-    pubsubTopics,
-    shardInfo
+    }
   };
 
+  if (shardInfo) {
+    waku_options.shardInfo = shardInfo;
```
Comment (adklempner) [inline packages/tests/tests/filter/utils.ts:99]: nit: replace snake case with camel case

**Review by adklempner — APPROVED** [review]

File: packages/sdk/src/create/libp2p.ts [hunk 4]
```diff
@@ -128,6 +117,37 @@ export async function createLibp2pAndUpdateOptions(
...
+  }
+
+  const shardInfo = options.shardInfo
+    ? ensureShardingConfigured(options.shardInfo)
+    : undefined;
+
+  options.pubsubTopics = shardInfo?.pubsubTopics ??
+    options.pubsubTopics ?? [DefaultPubsubTopic];
+
+  return shardInfo?.shardInfo;
```
[the remark this answers is not in the document]
Comment (weboko) [reply packages/sdk/src/create/libp2p.ts:148]: as I mentioned here - I don't think it is improvement for readability

```ts
if (shardInfo?.pubsubTopics) {
  options.pubsubTopics = shardInfo.pubsubTopics;
} else if (!options.pubsubTopics) {
  options.pubsubTopics = [DefaultPubsubTopic];
}

if (shardInfo?.shardInfo) {
  return shardInfo.shardInfo;
} 

return undefined;
```

again, I think this area should be re-worked and here is the task https://github.com/waku-org/js-waku/issues/2073

File: packages/sdk/src/create/libp2p.ts [hunk 5]
```diff
@@ -128,6 +117,37 @@ export async function createLibp2pAndUpdateOptions(
   return libp2p;
 }
 
+function configureNetworkOptions(
+  options: CreateWakuNodeOptions
+): ShardInfo | undefined {
```
[the remark this answers is not in the document]
Comment (weboko) [reply packages/sdk/src/create/libp2p.ts:122]: because this is how types are defined and `shardInfo` can be missing in some cases

**Pull request merged**